### PR TITLE
Add Material 3 `Slider` example

### PR DESCRIPTION
**Pull request by TahaTesser**
Part of https://github.com/flutter/flutter/issues/101048

This PR adds Material 3 `Slider` example since [Material 3 `Slider`](https://github.com/flutter/flutter/issues/111451)  landed.

Screenshot 2022-11-18 at 12 14 07

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Review by Piinks — APPROVED** [review]
LGTM-stamp

File: examples/api/test/material/slider/slider.1_test.dart [hunk 1]
```diff
@@ -7,29 +7,23 @@ import 'package:flutter_api_samples/material/slider/slider.1.dart' as example;
 import 'package:flutter_test/flutter_test.dart';
 
 void main() {
-  testWidgets('Slider shows secondary track', (WidgetTester tester) async {
+  testWidgets('Slider can change value', (WidgetTester tester) async {
```
Comment (Piinks) [inline examples/api/test/material/slider/slider.1_test.dart:10]:
```suggestion
  testWidgets('Slider can change its value', (WidgetTester tester) async {
```

Just kicking CI

Comment (Piinks) [reply examples/api/test/material/slider/slider.1_test.dart:10]: Oh I can't make a commit on this. Let em see if I can trigger a rerun on the failing check

**Comment by Piinks** [comment]
This might need to be rebased

File: examples/api/test/material/slider/slider.1_test.dart (hunk 1 above)
Comment (TahaTesser) [reply examples/api/test/material/slider/slider.1_test.dart]: Done :)

**Pull request merged**